packages/slice-machine/src/components/VideoPopover/VideoPopover.tsx

```diff
@@ -77,77 +159,24 @@ function useTimerWhen(condition: boolean, fn: () => void, delay: number) {
   }, [condition, fn, delay, timer, setTimer]);
 }
 
-type VideoPopoverProps = PropsWithChildren<{
-  open: boolean;
-  onClose: () => void;
-  onPlay?: () => void;
-  publicId: string;
-  cloudName: string;
-  delay?: number;
-  thumbnail?: string;
-  side?: HoverCardProps["side"];
-  sideOffset?: HoverCardProps["sideOffset"];
-  arrowSize?: number;
-  align?: HoverCardProps["align"];
-  alignOffset?: HoverCardProps["alignOffset"];
-}>;
-
-export const VideoPopover: FC<VideoPopoverProps> = ({
-  children,
-  onClose,
-  onPlay,
-  open,
-  delay = 5000,
-  publicId,
-  cloudName,
-  thumbnail,
-  side,
-  sideOffset,
-  arrowSize,
-  align,
-  alignOffset,
-}) => {
-  const [isOpen, setOpen] = useState<boolean>(false);
-
-  const handleClose = useCallback(() => {
-    onClose();
-    setOpen(false);
-  }, [onClose, setOpen]);
-
-  const handleDefaultOpening = useCallback(
-    () => setOpen((wasOpen) => wasOpen === false),
-    [setOpen]
-  );
-
-  useTimerWhen(open, handleDefaultOpening, delay);
-
-  const handleChange = (value: boolean) => {
-    if (value === false) {
-      handleClose();
-    } else if (value === true && open === true) {
-      setOpen(value);
+function useOnClickOutside<T extends HTMLElement = HTMLElement>(
```

Comment:
Following your question, yes, I would put this in our hooks folder.

Reply:
there's a similar hook in `usehooks-ts` ... not sure if it's worth inistalling the whole library for just a few hooks :/

Reply:
No don't think so, putting it in hooks folder is great already for me